src/libdrakvuf/drakvuf.c

```diff
@@ -879,3 +879,37 @@ bool drakvuf_set_vcpu_gprs(drakvuf_t drakvuf, int vcpu, registers_t* regs)
 
     return xen_set_vcpu_ctx(drakvuf->xen, drakvuf->domID, vcpu, &ctx);
 }
+
+bool drakvuf_enable_ipt(drakvuf_t drakvuf, int vcpu, uint8_t** buf, uint64_t* size)
+{
+    int rc = xen_enable_ipt(drakvuf->xen, drakvuf->domID, vcpu, &drakvuf->ipt_state[vcpu]);
+
+    if (rc != 0)
+    {
+        *buf = drakvuf->ipt_state[vcpu].buf;
+        *size = drakvuf->ipt_state[vcpu].size;
+        return true;
+    }
+
+    return false;
+}
+
+bool drakvuf_get_ipt_offset(drakvuf_t drakvuf, int vcpu, uint64_t* offset, uint64_t* last_offset)
+{
+    int rc = xen_get_ipt_offset(drakvuf->xen, drakvuf->domID, vcpu, &drakvuf->ipt_state[vcpu]);
+
+    if (rc != 0)
+    {
+        *offset = drakvuf->ipt_state[vcpu].offset;
+        *last_offset = drakvuf->ipt_state[vcpu].last_offset;
+        return true;
+    }
+
+    return false;
+}
+
+bool drakvuf_disable_ipt(drakvuf_t drakvuf, int vcpu)
+{
+    int rc = xen_disable_ipt(drakvuf->xen, drakvuf->domID, vcpu, &drakvuf->ipt_state[vcpu]);
```

Comment:
Same, no need for `int rc`, you can just return `xen_disable_ipt` right away, it's already effectively a bool.